File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        n = len(nums)
        inc, dec, ans = 1, 1, 1

        for i in range(1, n):
            if nums[i - 1] < nums[i]:
                dec = 1
                inc += 1
            elif nums[i - 1] > nums[i]:
                inc = 1
                dec += 1
            else:
                inc = 1
                dec = 1

            ans = max(ans, inc, dec)

        return ans

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int n = nums.size();
        int inc = 1, dec = 1, ans = 1;

        for (int i = 1; i < n; i++) {
            if (nums[i - 1] < nums[i]) {
                dec = 1;
                inc += 1;
            } else if (nums[i - 1] > nums[i]) {
                inc = 1;
                dec += 1;
            } else {
                inc = 1;
                dec = 1;
            }

            ans = std::max(ans, std::max(inc, dec));
        }

        return ans;
    }
};